src/lsstseries/ensemble.py

```diff
@@ -603,37 +677,64 @@ def from_parquet(
             self._err_col = err_col
         if band_col is not None:
             self._band_col = band_col
+        if provenance_col is not None:
+            self._provenance_col = provenance_col
 
         if additional_cols:
             columns = None  # None will prompt read_parquet to read in all cols
+
         else:
             columns = [self._time_col, self._flux_col, self._err_col, self._band_col]
+            if self._provenance_col is not None:
+                columns.append(self._provenance_col)
 
         # Read in the source parquet file(s)
         self._source = dd.read_parquet(
             source_file, index=self._id_col, columns=columns, split_row_groups=True
         )
 
-        if npartitions and npartitions > 1:
-            self._source = self._source.repartition(npartitions=npartitions)
-        elif partition_size:
-            self._source = self._source.repartition(partition_size=partition_size)
-
         if object_file:  # read from parquet files
-            # Read in the object file
-            file_table = dd.read_parquet(object_file, index=self._id_col, split_row_groups=True)
+            # Read in the object file(s)
+            self._object = dd.read_parquet(object_file, index=self._id_col, split_row_groups=True)
+
+            # Handle nobs_band columns
+            if nobs_cols is not None:
```

Comment:
Are these going to be in enough files to justify reading them? Alternatively we could just always auto generate them with sync tables. I understand that we can get it for free it is already in the parquet file, I'm just trying to balance that efficiency vs specializing too much of the code.

Reply:
So far we know they are available in ZTF and Panstarrs, I wasn't able to find them in the LSST DP0.2 schema, but there was at least a column for the total nobs it looked like. So potentially?

Reply:
We can keep it in the code for now, but I'm worried about adding too much column-specific logic.